src/core/component.js

```diff
@@ -212,7 +215,7 @@ module.exports.registerComponent = function (name, definition) {
   });
 
   if (components[name]) {
-    throw new Error('The component ' + name + ' has been already registered');
+    warn('The component ' + name + ' has been already registered');
```

Comment:
I think this makes more sense to be an error than a warning. If a component is registered twice is a big problem.